File: ОП/Lab 6/Lab 6 C++/Lab6/Lab6.cpp
#include <iostream>
using namespace std;

int func(int n)
{
	if (n>=10) {
		do {
			n /= 10;
		} while (n>=10);
		return n;
	}
	else {
		return n;
	}
}

int main()
{
	int a, b, c, sum;
	cout << "Enter a, b, c ∈ [1; +inf)" << endl;
	cin >> a >> b >> c;
	if (a > 0 && b > 0 && c > 0) {
		sum = func(a) + func(b) + func(c);
		cout << "Sum of first digits of entered numbers is:" << endl;
		cout << sum << endl;
	}
	else {
		cout << "One number (or more) doesn't belong to the required range. Try again." << endl;
	}
	
	system("pause");
	return 0;
}

File: ОП/Lab 6/Lab 6 Python/Lab6.py
def func(n):
    if n >=10:
        while n >= 10:
            k = n
            n /= 10
        mod = (k % 10) / 10
        func_res = n - mod
        return func_res
    else:
        func_res=n
        return func_res
        
print ("Enter a, b, c ∈ [1; +inf)")
a = int(input())
b = int(input())
c = int(input())

if a >= 1 and b >= 1 and c >= 1:
    res = func(a) + func(b) + func(c)
    print ("Sum of first digits of entered numbers is:", res)
else:
    print ("One number (or more) doesn't belong to the required range. Try again.")

input("Press Enter to continue.")